suites/src/main/java/com/hedera/block/suites/persistence/positive/PositiveDataPersistenceTests.java

```diff
@@ -16,12 +16,95 @@
 
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.hedera.block.simulator.BlockStreamSimulatorApp;
+import com.hedera.block.simulator.BlockStreamSimulatorInjectionComponent;
+import com.hedera.block.simulator.DaggerBlockStreamSimulatorInjectionComponent;
 import com.hedera.block.suites.BaseSuite;
+import java.io.IOException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.Container;
 
-/** Positive Data Persistence Tests */
+/**
+ * Test class for verifying the positive scenarios for data persistence.
+ *
+ * <p>Inherits from {@link BaseSuite} to reuse the container setup and teardown logic for the Block
+ * Node.
+ */
 @DisplayName("Positive Data Persistence Tests")
 public class PositiveDataPersistenceTests extends BaseSuite {
+    private final String[] GET_BLOCKS_COMMAND = new String[] {"ls", "data", "-1"};
+
+    private Thread simulatorThread;
+
     /** Default constructor for the {@link PositiveDataPersistenceTests} class. */
     public PositiveDataPersistenceTests() {}
+
+    @BeforeEach
+    void setupEnvironment() throws IOException {
+        BlockStreamSimulatorInjectionComponent DIComponent =
+                DaggerBlockStreamSimulatorInjectionComponent.factory()
+                        .create(loadSimulatorDefaultConfiguration());
+
+        blockStreamSimulatorApp = DIComponent.getBlockStreamSimulatorApp();
+        simulatorThread = setupSimulatorThread(blockStreamSimulatorApp);
+    }
+
+    @AfterEach
+    void teardownEnvironment() {
+        simulatorThread.interrupt();
+    }
+
+    /**
+     * Verifies that block data is saved in the correct directory by comparing the count of saved
+     * blocks before and after running the simulator. The test asserts that the number of saved
+     * blocks increases after the simulator runs.
+     *
+     * @throws IOException if an I/O error occurs during execution in the container
+     * @throws InterruptedException if the thread is interrupted while sleeping or executing
+     *     commands
+     */
+    @Test
+    public void verifyBlockDataSavedInCorrectDirectory() throws InterruptedException, IOException {
+        String savedBlocksFolderBefore = getContainerCommandResult(GET_BLOCKS_COMMAND);
+        int savedBlocksCountBefore = getSavedBlocksCount(savedBlocksFolderBefore);
+
+        simulatorThread.start();
+        Thread.sleep(5000);
+        blockStreamSimulatorApp.stop();
+
+        String savedBlocksFolderAfter = getContainerCommandResult(GET_BLOCKS_COMMAND);
+        int savedBlocksCountAfter = getSavedBlocksCount(savedBlocksFolderAfter);
+
+        assertTrue(savedBlocksFolderBefore.isEmpty());
+        assertFalse(savedBlocksFolderAfter.isEmpty());
+        assertTrue(savedBlocksCountAfter > savedBlocksCountBefore);
+    }
+
```

Comment:
Rather than spin up new Threads individually (expensive). Consider using `private ExecutorService executorService = Executors.newFixedThreadPool(8)` and managing simulator instances from that API with something like: `executorService.execute(blockStreamSimulatorAppInstance::start);`. The `ExecutorService` gives you a number of thread lifecycle management methods for free.

Reply:
Thanks, used the apporach you proposed, I agree with you on your argument. Also tests and how they setup everything might be changed in the future, idk.